sysutils/wxHexEditor/files/0001-Fix-build-for-wxWidgets-v3.1.2.patch

```diff
@@ -0,0 +1,40 @@
+From ebe2449fac22089825d124935a215fd1c0739403 Mon Sep 17 00:00:00 2001
+From: GT <george@tsimperopoulos.com>
+Date: Tue, 10 Sep 2019 23:52:19 +0100
+Subject: [PATCH] Fix build for wxWidgets v3.1.2
+
+---
+ src/HexDialogs.cpp                  | 2 +-
+ src/HexEditorCtrl/HexEditorCtrl.cpp | 6 +++---
+ 2 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git src/HexDialogs.cpp src/HexDialogs.cpp
+index 260571b..9c312ba 100644
+--- src/HexDialogs.cpp
++++ src/HexDialogs.cpp
+@@ -420,7 +420,7 @@ void FindDialog::OnChar( wxKeyEvent& event ){
+ 	}
+ 
+ void FindDialog::EventHandler( wxCommandEvent& event ){
+-	WX_CLEAR_ARRAY(parent->HighlightArray )
```

Comment:
What was going on here? I assume this compiled before? How?

Reply:
This is a new port. What do you mean it was compiler before? :) 

Also, the patch is from upstream. (As you can see, it is not by me.)

Reply:
I assume that upstream this code had to have compiled. How did this compile at all if the patch is needed?

Reply:
With an earlier version of `wxWidgets` (which is still used for older OS here). My patch is literally a commit from upstream. Why is this suddenly a problem?

Even if it _can_ work with old wxWidgets on Sonoma (I am not entirely sure it will), why would we want to use old wxWidgets on new macOS? What is the gain?

If you insist, of course, I can try setting it back to old version and drop the patch.

We still need two versions of wxWidgets though.

Reply:
It's a problem because the patch is weird, and I want to understand how it came to exist at all. If it's necessary, the pre-patch code could not have compiled on any platform whatsoever, which makes it odd that it would have been released by upstream.

My job is to make sure we commit reasonable code to the repository. When something gets weird, I want to understand it.

Reply:
I am not an expert in wxWidgets API. Without a patch it failed with wxWidgets 3.2 with:
```
:info:build In file included from src/HexDialogs.h:30:
:info:build src/HexEditorCtrl/HexEditorCtrl.h:130:8: warning: 'HexEditorCtrl::SetFont' hides overloaded virtual function [-Woverloaded-virtual]
:info:build                 void SetFont( );
:info:build                      ^
:info:build /opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.1/include/wx-3.2/wx/osx/window.h:87:18: note: hidden overloaded virtual function 'wxWindow::SetFont' declared here: different number of parameters (1 vs 0)
:info:build     virtual bool SetFont( const wxFont &font ) wxOVERRIDE;
:info:build                  ^
:info:build src/HexDialogs.cpp:423:41: error: expected ';' after expression
:info:build         WX_CLEAR_ARRAY(parent->HighlightArray )
:info:build                                                ^
:info:build                                                ;
```
Turned out, the error was already dealt with by upstream, so I borrowed a patch.

P. S. You seem to miss that we are talking about several versions of wxWidgets. Why would the code for them have to be absolutely identical? It is not the case with Qt or GTK, for example.
And if you wish, you could build locally and verify it works.